user/session.go

```diff
@@ -491,18 +492,26 @@ type EndpointRateLimitInfo struct {
 }
 
 // RateLimitInfo returns EndpointRateLimitInfo for endpoint rate limiting.
-func (es Endpoints) RateLimitInfo(method string, path string) (*EndpointRateLimitInfo, bool) {
+func (es Endpoints) RateLimitInfo(method string, reqEndpoint string) (*EndpointRateLimitInfo, bool) {
 	if len(es) == 0 {
 		return nil, false
 	}
 
 	for _, endpoint := range es {
-		asRegex, err := regexp.Compile(endpoint.Path)
+		url := endpoint.Path
```

Comment:
I think this needs some refactoring for reuse, extract the inner loop to a new function:

*Endpoint.Match(reqPath string) (bool, error);

Generally the logic is duplicated with allowedURLs, a httputil addition is welcome, perhaps also a Match() func there that can be used from both places.